image/manifest_test.go

```diff
@@ -48,3 +64,60 @@ func TestUnpackLayerDuplicateEntries(t *testing.T) {
 		t.Fatalf("Expected to fail with duplicate entry, got %v", err)
 	}
 }
+func TestUnpackLayerRemovePartialyUnpackedFile(t *testing.T) {
+	tmp1, err := ioutil.TempDir("", "test-layer")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmp1)
+	err = os.MkdirAll(filepath.Join(tmp1, "blobs", "sha256"), 0700)
+	if err != nil {
+		t.Fatal(err)
+	}
+	tarfile := filepath.Join(tmp1, "blobs", "sha256", "test.tar")
```

Comment:
`test.tar` doesn't look like a hex hash ;).  I'd rather build this layer outside of `blobs/sha256` and then rename it into position once we've completed it and can get a content-addressable hash.

And “create image-layout fodder for testing” is likely to be useful stuff in lots of places.  Can we make layer-generator, etc. helpers instead of baking it into `TestUnpackLayerRemovePartialyUnpackedFile`.  I'm missing #159 and 87d6a53 ;).